contracts/pcv/compound/ERC4626CErc20Wrapper.sol

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import "./CErc20.sol";
+import "../../external/IERC4626.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @title ERC4626 wrapper around a Cerc20
+/// @author Fei Protocol
+contract ERC4626CErc20Wrapper is IERC4626, ERC20 {
+    using SafeERC20 for IERC20;
+    using SafeERC20 for CErc20;
+
+    /// @notice CErc20 token reference
+    CErc20 public immutable cToken;
+
+    /// @notice The address of the underlying ERC20 token used for
+    /// the Vault for accounting, depositing, and withdrawing.
+    IERC20 public immutable cTokenUnderlying;
+
+    /*///////////////////////////////////////////////////////////////
+                Constructor
+    //////////////////////////////////////////////////////////////*/
+    /// @notice ERC4626CErc20Wrapper constructor
+    /// @param _cToken Compound cToken to wrap
+    /// @param name ERC20 name of the vault shares token
+    /// @param symbol ERC20 symbol of the vault shares token
+    constructor(
+        address _cToken,
+        string memory name,
+        string memory symbol
+    ) ERC20(name, symbol) {
+        require(CErc20(_cToken).isCToken(), "ERC4626Cerc20Wrapper: not a cToken");
+        require(!CErc20(_cToken).isCEther(), "ERC4626Cerc20Wrapper: cEther not supported");
+        cToken = CErc20(_cToken);
+        cTokenUnderlying = IERC20(CErc20(cToken).underlying());
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                ERC4626 Vault properties
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The address of the underlying ERC20 token used for
+    /// the Vault for accounting, depositing, and withdrawing.
+    function asset() external view returns(address) {
+        return address(cTokenUnderlying);
+    }
+
+    /// @notice Total amount of the underlying asset that
+    /// is "managed" by Vault.
+    function totalAssets() external view returns(uint256) {
+        return cToken.getCash() + cToken.totalBorrows();
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                ERC4626 Deposit/Withdrawal logic
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Mints `shares` Vault shares to `receiver` by
+    /// depositing exactly `amount` of underlying tokens.
+    function deposit(uint256 assets, address receiver) external returns(uint256 shares) {
+        // transfer user tokens to self
+        cTokenUnderlying.safeTransferFrom(msg.sender, address(this), assets);
+
+        // mint cTokens
+        cTokenUnderlying.approve(address(cToken), assets);
+        uint256 balanceBefore = cToken.balanceOf(address(this));
```

Comment:
Should not read an updated balance, instead there should be a sweep method

Reply:
yes, changing